consul/config/bootstrap-consul.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+METADATA_URL="http://metadata.google.internal/computeMetadata/v1"
+INSTANCE_METADATA_URL="$METADATA_URL/instance"
+INSTANCE_LIST_METADATA="consul-members"
+WAIT_TIME_SECONDS=120
+
+function get_instance_metadata_attribute() {
+  local name="$1"
+  local value=$(curl -s -f -H "Metadata-Flavor: Google" \
+                     $INSTANCE_METADATA_URL/attributes/$name)
+  if [[ $? -eq 0 ]]; then
+    echo "$value"
+  else
+    echo ""
+  fi
+}
+
+function get_instance_list_from_metadata() {
+  get_instance_metadata_attribute $INSTANCE_LIST_METADATA
+}
+
+function get_instance_list_from_mig() {
+  # projects/<projectid>/zones/<zone>/instanceGroupManagers/<igm>
+  createduri=$(get_instance_metadata_attribute "created-by")
+
+  if [ -z $createduri ]
+  then
+    echo "Failed to retrieve who created" $HOSTNAME
+    echo "Make sure it is running in a managed instance group"
+    exit 1
+  fi
+
+  IFS='/' read -r -a createdlist <<< "$createduri"
+
+  mig=${createdlist[-1]}
+  zone=${createdlist[-3]}
+
+  gcloud compute instance-groups list-instances $mig --zone $zone --format='value(instance)'
+}
+
+instances=$(get_instance_list_from_metadata)
+
+if [ -z "$instances" ]
+then
+  echo "Consul instances not found in instance metadata, attempting to join managed instance group."
+  instances=$(get_instance_list_from_mig)
+fi
+
+if [ -z "$instances" ]
+then
+  echo "Failed to retrieve consul instances" 
+  exit 1
+fi
+
+echo "Waiting" $WAIT_TIME_SECONDS "before joining nodes" $instances
+
+sleep $WAIT_TIME_SECONDS
```

Comment:
if you are waiting on a condition, you should try to test for that condition rather than assume it will be ready on an upper bound. I'd be aggressive testing earlier since presumably this is an upper bound that you have expected to nearly always exceed the actual required time.